README.rst

```diff
@@ -1,414 +1,133 @@
-
-    class FooSettings(Settings):
-        schema = {
-            'serializer': conformity.fields.Dictionary({
-                'path': conformity.fields.UnicodeString(),
-            }),
-        }
-
-        def convert_serializer(self, value):
-            if "object" not in value:
-                try:
-                    value["object"] = self.resolve_python_path(value["path"])
-                except ImportError:
-                    raise self.ImproperlyConfigured(
-                        "Could not resolve path '{}' for configuration:\n{}".format(value["path"], value))
-            return value
-
-    my_settings = FooSettings({
-        'serializer': {
-            'path': 'pysoa.common.serializer:JSONSerializer'
-        }
-    })
-
-    my_settings['serializer']['object']
-    > pysoa.common.serializer.json_serializer.JSONSerializer
-
-  The method ``resolve_python_path`` takes care of the import for you. The ``Settings`` class also provides a convenience method called ``standard_convert_path`` that converts a value exactly as above, so ``convert_serializer`` could simply be written as::
-
-    def convert_serializer(self, value):
-        return self.standard_convert_path(value)
-
-
-Settings subclasses
-*******************
-
-``common.settings.SOASettings``:
+--------
 
-- Schema:
+Both client and server also use a dict-based settings system, with a
+`conformity <https://github.com/eventbrite/conformity>`_-defined schema to ensure
+that whatever is passed in is valid (this is extensible by service implementations
+if they have special settings they need set).
 
-  + ``transport``: Import path and keyword args for a ``Transport`` class.
+The server also has an integration mode with Django where it will read its
+settings from ``django.conf.settings.SOA_SERVER_SETTINGS`` for both running and
+for tests, which allows easy integration of Django models and application
+logic into services (we make heavy use of the Django ORM).
 
-  + ``serializer``: Import path and keyword args for a ``Serializer`` class.
 
-  + ``middleware``: List of dicts containing import path and keyword args for a ``ClientMiddleware`` or ``ServerMiddleware`` class.
+Testing
+-------
 
-Both the ``client`` and ``server`` modules implement their own subclasses that inherit from ``SOASettings``. Developers implementing ``Client`` or ``Server`` subclasses may wish to subclass the respective settings class in order to alter or extend the schema or defaults.
+Services can be tested using standard unit tests and either by calling the
+actions directly (after all, they are just callables), or if a run through the
+server machinery is desired, using the ``ServerTestCase`` base class which takes
+care of setting up local transports for you.
```

Comment:
We should mention that there is a StubClient as well, for testing code that calls your service.